ambry-commons/src/main/java/com.github.ambry.commons/BlobId.java

```diff
@@ -162,18 +307,52 @@ public boolean equals(Object o) {
     if (!version.equals(blobId.version)) {
```

Comment:
why not simply write `return version.equals(blobId.version) && partitionId.equals(blobId.partitionId) && .... && uuid.equals(blobId.uuid)`?